### ISSUE TITLE
Embedding of python keras does not update weights

### ISSUE THREAD

**Issue by songhappy** [intel-analytics/analytics-zoo#1327]
np.random.seed(1337)  # for reproducibility

    def test_embedding(self):
        def gen_sample(user_num, item_num, class_num):
            user_id = random.randint(1, user_num)
            item_id = random.randint(1, item_num)
            rating = random.randint(1, class_num)
            sample = Sample.from_ndarray(np.array([user_id, item_id]), np.array([rating]))
            return sample
        data = self.sc.parallelize(range(0, 50)) \
            .map(lambda i: gen_sample(10, 10, 2))

        def build_model(user_count, item_count, mf_embed, class_num):
            input = Input(shape=(2,))
            user_flat = Flatten()(Select(1, 0)(input))
            item_flat = Flatten()(Select(1, 1)(input))
            mf_user_embed = Embedding(user_count + 1, mf_embed, init="uniform")(user_flat)
            mf_item_embed = Embedding(item_count + 1, mf_embed, init="uniform")(item_flat)
            mf_user_flatten = Flatten()(mf_user_embed)
            mf_item_flatten = Flatten()(mf_item_embed)
            mf_latent = merge(inputs=[mf_user_flatten, mf_item_flatten], mode="mul")
            linear_last = Dense(class_num, activation="softmax")(mf_latent)
            model = Model(input, linear_last)
            return model

        model = build_model(10, 10, 4, 2)
        print("----------before fitting--------------")
        print(model.get_weights())
        model.summary()
        model.compile(optimizer="adam",
                      loss=SparseCategoricalCrossEntropy(zero_based_label=False),
                      metrics=['accuracy'])
        tmp_log_dir = create_tmp_path()
        model.set_tensorboard(tmp_log_dir, "training_test")
        for i in range(1, 10):
            model.fit(data, nb_epoch=1, batch_size=32, validation_data=data)
            print("----------inb_epoch: " + str(i) + "inside fitting--------------")
            print(model.get_weights())

        print("----------after fitting--------------")
        print(model.get_weights())

----------before fitting--------------
[array([[-0.03849747, -0.0368647 ,  0.01192915,  0.03788165],
       [-0.00032834,  0.04914744,  0.02832074, -0.03352324],
       [ 0.03727734, -0.0452476 ,  0.0027833 , -0.01544324],
       [ 0.00589378,  0.0479706 ,  0.01913423, -0.00308059],
       [ 0.02953278,  0.0434649 ,  0.00280901,  0.02037395],
       [ 0.04640509, -0.04021541, -0.02038394,  0.04990844],
       [-0.00996695, -0.03170951,  0.04927721,  0.01269252],
       [-0.02392467,  0.01219499, -0.03246098, -0.0250195 ],
       [-0.02552338,  0.03102412,  0.01886255,  0.03725457],
       [ 0.00628602,  0.03142597, -0.03670193,  0.02725225],
       [ 0.04088141, -0.04817726,  0.00026879,  0.0202721 ]],
      dtype=float32), array([[ 0.0156287 , -0.0435855 , -0.0431373 , -0.04085271],
       [-0.01837587,  0.03327176,  0.00665995, -0.03911637],
       [ 0.00161683, -0.01358181, -0.02298188,  0.02178464],
       [ 0.01035214,  0.01203047, -0.00624319, -0.04411566],
       [-0.01154471,  0.00291077, -0.03935052, -0.0079064 ],
       [ 0.01938303, -0.03830246,  0.00110278, -0.01886901],
       [ 0.01260849,  0.03150135, -0.0396903 ,  0.03751658],
       [ 0.01150031,  0.00862092, -0.01816351,  0.0478849 ],
       [-0.00016531,  0.03888714, -0.03661468,  0.03199743],
       [-0.01205093, -0.03124828, -0.0004036 ,  0.02798083],
       [ 0.03185649,  0.00547173, -0.04211359, -0.04633185]],
      dtype=float32), array([[ 0.6034    , -0.22266531,  0.43221495,  0.64649326],
       [ 0.23195474, -0.618709  , -0.8666277 , -0.6761056 ]],
      dtype=float32), array([0., 0.], dtype=float32)]
----------inb_epoch: 1inside fitting--------------
[array([[-0.03849747, -0.0368647 ,  0.01192915,  0.03788165],
       [ 0.00166367,  0.05113861,  0.02632342, -0.03273654],
       [ 0.03543155, -0.04325175,  0.00111029, -0.01587206],
       [ 0.00589378,  0.0479706 ,  0.01913423, -0.00308059],
       [ 0.02753977,  0.04147048,  0.0047987 ,  0.02237427],
       [ 0.04457942, -0.04206582, -0.0185565 ,  0.05174152],
       [-0.00800039, -0.02975341,  0.04731075,  0.0107281 ],
       [-0.02582743,  0.0102755 , -0.03422521, -0.02309783],
       [-0.02356081,  0.03300531,  0.01686242,  0.0392104 ],
       [ 0.00436414,  0.03336212, -0.03479563,  0.0253245 ],
       [ 0.038889  , -0.04619254, -0.00172338,  0.01827567]],
      dtype=float32), array([[ 0.0156287 , -0.0435855 , -0.0431373 , -0.04085271],
       [-0.01709915,  0.03496635,  0.00519395, -0.03715455],
       [ 0.00161683, -0.01358181, -0.02298188,  0.02178464],
       [ 0.00835562,  0.01144413, -0.00427889, -0.04610576],
       [-0.0095531 ,  0.0009198 , -0.03893882, -0.00591263],
       [ 0.01738524, -0.03630403, -0.00085736, -0.01686876],
       [ 0.01088922,  0.03348791, -0.03769909,  0.03551996],
       [ 0.00963741,  0.01049609, -0.01630107,  0.04975938],
       [-0.00016531,  0.03888714, -0.03661468,  0.03199743],
       [-0.01396088, -0.02932618, -0.00231821,  0.02605897],
       [ 0.02993141,  0.0062    , -0.04011263, -0.04801785]],
      dtype=float32), array([[ 0.6014569 , -0.22068135,  0.4302163 ,  0.6482743 ],
       [ 0.23389785, -0.62069297, -0.86462903, -0.67788666]],
      dtype=float32), array([ 0.00197651, -0.00197651], dtype=float32)]
----------inb_epoch: 2inside fitting--------------
[array([[-0.03849747, -0.0368647 ,  0.01192915,  0.03788165],
       [ 0.00122857,  0.05307188,  0.02439631, -0.03081003],
       [ 0.03343225, -0.04128456, -0.00081254, -0.01394405],
       [ 0.00788924,  0.04975321,  0.01715889, -0.00108672],
       [ 0.02753977,  0.04147048,  0.0047987 ,  0.02237427],
       [ 0.04656595, -0.04007281, -0.02054538,  0.04974944],
       [-0.00800039, -0.02975341,  0.04731075,  0.0107281 ],
       [-0.02582743,  0.0102755 , -0.03422521, -0.02309783],
       [-0.02538447,  0.03491043,  0.01877434,  0.03728777],
       [ 0.00636321,  0.03536176, -0.03283427,  0.02332584],
       [ 0.04087318, -0.04433448, -0.00361899,  0.02024177]],
      dtype=float32), array([[ 0.0156287 , -0.0435855 , -0.0431373 , -0.04085271],
       [-0.01908451,  0.03296771,  0.00319721, -0.03519147],
       [ 0.00352495, -0.01549485, -0.02489194,  0.01986409],
       [ 0.00835562,  0.01144413, -0.00427889, -0.04610576],
       [-0.00758573, -0.00104646, -0.03722923, -0.00787392],
       [ 0.01538649, -0.0343054 , -0.00258998, -0.01487167],
       [ 0.01275602,  0.03161933, -0.03587391,  0.03365497],
       [ 0.0116323 ,  0.00849845, -0.01825657,  0.05175385],
       [ 0.00161473,  0.04062352, -0.03463366,  0.03058448],
       [-0.01595378, -0.03132599, -0.00031935,  0.02406055],
       [ 0.03191882,  0.00420854, -0.04209948, -0.04602586]],
      dtype=float32), array([[ 0.603412  , -0.22267057,  0.42943835,  0.64628786],
       [ 0.23194277, -0.6187038 , -0.8638511 , -0.6759002 ]],
      dtype=float32), array([ 0.00397636, -0.00397636], dtype=float32)]
...
----------after fitting--------------
[array([[-0.03849747, -0.0368647 ,  0.01192915,  0.03788165],
       [-0.00641369,  0.05906224,  0.02647505, -0.04085837],
       [ 0.03737217, -0.04135116, -0.00094848, -0.01008746],
       [ 0.00938525,  0.04892112,  0.01945207, -0.00605041],
       [ 0.02554332,  0.04347461, -0.00087687,  0.02437269],
       [ 0.04688632, -0.04006661, -0.01919396,  0.04564625],
       [-0.00432807, -0.02571856,  0.04330176,  0.01517777],
       [-0.02446537,  0.01277999, -0.03712359, -0.01626698],
       [-0.02542878,  0.02705956,  0.01486829,  0.04132487],
       [ 0.00689798,  0.02858469, -0.03372477,  0.02439529],
       [ 0.04286448, -0.03843561, -0.00596729,  0.01843144]],
      dtype=float32), array([[ 0.0156287 , -0.0435855 , -0.0431373 , -0.04085271],
       [-0.0190528 ,  0.0328794 ,  0.00709456, -0.04208492],
       [-0.00175739, -0.01412105, -0.0251886 ,  0.02774984],
       [ 0.01238925,  0.01152292, -0.00429183, -0.04612502],
       [-0.01394872,  0.00281447, -0.03803135, -0.00798303],
       [ 0.01773501, -0.03663695, -0.00094753, -0.01654346],
       [ 0.01077283,  0.02973319, -0.03672308,  0.03945017],
       [ 0.00957858,  0.00255712, -0.01668424,  0.04681882],
       [ 0.00277444,  0.04054784, -0.03483538,  0.03462624],
       [-0.0200343 , -0.03125756, -0.00325603,  0.02801169],
       [ 0.03212494,  0.00034936, -0.03813751, -0.04617977]],
      dtype=float32), array([[ 0.6086017 , -0.22502665,  0.42770624,  0.65370643],
       [ 0.226753  , -0.6163476 , -0.862119  , -0.6833188 ]],
      dtype=float32), array([ 0.0099994, -0.0099994], dtype=float32)]
                        [100%]

**Comment by songhappy** [comment]
false alarm, it is updated.
